superset/utils/mock_data.py

```diff
@@ -68,7 +70,9 @@
 
 
 # pylint: disable=too-many-return-statements, too-many-branches
-def get_type_generator(sqltype: sqlalchemy.sql.sqltypes) -> Callable[[], Any]:
+def get_type_generator(
+    sqltype: sqlalchemy.sql.sqltypes, group_by: bool = False
+) -> Callable[[], Any]:
```

Comment:
@hugh can you give more details about why you need to pass `group_by` here? Can we achieve this some other way?

A group by is not an intrinsic concept of a table, so passing it to a function that populates a table with random data is confusing.

If you want the column name to have a single character and can't have that because of line 94 (`length = max(8, length)`), I think it's better to create your own type:

```python
class SingleChar(sqlalchemy.sql.sqltypes.String):
    pass
```

Then modify the function to check for your type before checking for `sqlalchemy.sql.sqltypes.String`:

```python
if isinstance(sqltype, sqlalchemy.sql.sqltypes.String):
    return lambda: random.choice(string.ascii_letters)

if isinstance(
        sqltype, (sqlalchemy.sql.sqltypes.VARCHAR, sqlalchemy.sql.sqltypes.String)
):
    ...
```

Then define your columns:

```python
from superset.utils.mock_data import SingleChar

column = {
    "name": "groupable",
    "type": SingleChar(),
    "nullable": False,
    "default": None,
    "autoincrement": "auto",
    "primary_key": 0,
}
```

Reply:
Another option would be passing a `cardinality` flag when generating a column.